rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/dap/RascalDebugEventTrigger.java

```diff
@@ -0,0 +1,108 @@
+package org.rascalmpl.vscode.lsp.dap;
+
+import io.usethesource.vallang.ISourceLocation;
+import org.eclipse.lsp4j.debug.StoppedEventArguments;
+import org.eclipse.lsp4j.debug.services.IDebugProtocolClient;
+import org.rascalmpl.debug.*;
+import org.rascalmpl.vscode.lsp.dap.breakpoint.BreakpointsCollection;
+
+/**
+ * This class handled events triggered by the Rascal interpreter (suspension, resume, ...)
+ */
+public class RascalDebugEventTrigger extends AbstractInterpreterEventTrigger {
+
+    private IDebugProtocolClient client;
+    private final SuspendedState suspendedState;
+    private final BreakpointsCollection breakpointsCollection;
+    private final DebugHandler debugHandler;
+
+    public RascalDebugEventTrigger(Object source, BreakpointsCollection breakpointsCollection, SuspendedState suspendedState, DebugHandler debugHandler) {
+        super(source);
+        this.breakpointsCollection = breakpointsCollection;
+        this.suspendedState = suspendedState;
+        this.debugHandler = debugHandler;
+    }
+
+    public void setDebugProtocolClient(IDebugProtocolClient client) {
+        this.client = client;
+    }
+
+    @Override
+    public void addRascalEventListener(IRascalEventListener listener) {
+        throw new UnsupportedOperationException("Unimplemented method 'addRascalEventListener'");
```

Comment:
are they not implemented, or shouldn't they be called? in the last case, update the message.